OrbitCaptureClient/include/OrbitCaptureClient/GpuQueueSubmissionProcessor.h

```diff
@@ -0,0 +1,110 @@
+  // use the `begin_capture_time_ns_` as an approximation for the begin CPU timestamp.
+  // This method updates this timestamp with the minimum of the current value and the given value.0
+  void UpdateBeginCaptureTime(uint64_t timestamp) {
+    begin_capture_time_ns_ = std::min(begin_capture_time_ns_, timestamp);
+  }
+
+ private:
+  [[nodiscard]] std::vector<orbit_client_protos::TimerInfo>
+  ProcessGpuQueueSubmissionWithMatchingGpuJob(
+      const orbit_grpc_protos::GpuQueueSubmission& gpu_queue_submission,
+      const orbit_grpc_protos::GpuJob& matching_gpu_job,
+      const absl::flat_hash_map<uint64_t, std::string>& string_intern_pool,
+      const std::function<uint64_t(const std::string& str)>&
+          get_string_hash_and_send_to_listener_if_necessary);
+  void ProcessGpuCommandBuffers(
+      const orbit_grpc_protos::GpuQueueSubmission& gpu_queue_submission,
+      const orbit_grpc_protos::GpuJob& matching_gpu_job,
+      const std::optional<orbit_grpc_protos::GpuCommandBuffer>& first_command_buffer,
+      uint64_t timeline_hash,
+      const std::function<uint64_t(const std::string& str)>&
+          get_string_hash_and_send_to_listener_if_necessary,
+      std::vector<orbit_client_protos::TimerInfo>* result);
+  void ProcessGpuDebugMarkers(
+      const orbit_grpc_protos::GpuQueueSubmission& gpu_queue_submission,
+      const orbit_grpc_protos::GpuJob& matching_gpu_job,
+      const std::optional<orbit_grpc_protos::GpuCommandBuffer>& first_command_buffer,
+      const std::string& timeline,
+      const absl::flat_hash_map<uint64_t, std::string>& string_intern_pool,
+      const std::function<uint64_t(const std::string& str)>&
+          get_string_hash_and_send_to_listener_if_necessary,
+      std::vector<orbit_client_protos::TimerInfo>* result);
+  [[nodiscard]] static std::optional<orbit_grpc_protos::GpuCommandBuffer> ExtractFirstCommandBuffer(
+      const orbit_grpc_protos::GpuQueueSubmission& gpu_queue_submission);
+  [[nodiscard]] const orbit_grpc_protos::GpuJob* FindMatchingGpuJob(
+      int32_t thread_id, uint64_t pre_submission_cpu_timestamp,
+      uint64_t post_submission_cpu_timestamp);
+  // Finds the GpuQueueSubmission that fully contains the given timestamp and happened on the given
+  // thread id. Returns `nullptr` if there is no such submission.
+  [[nodiscard]] const orbit_grpc_protos::GpuQueueSubmission* FindMatchingGpuQueueSubmission(
+      int32_t thread_id, uint64_t submit_time);
+  [[nodiscard]] bool HasUnprocessedBeginMarkers(int32_t thread_id,
+                                                uint64_t post_submission_timestamp) const;
+  void DecrementUnprocessedBeginMarkers(int32_t thread_id, uint64_t submission_timestamp,
+                                        uint64_t post_submission_timestamp);
+  void DeleteSavedGpuJob(int32_t thread_id, uint64_t submission_timestamp);
+  void DeleteSavedGpuSubmission(int32_t thread_id, uint64_t post_submission_timestamp);
```

Comment:
Add spaces between these methods. It's quite hard to read otherwise.

Reply:
Done.